scalafix-tests/integration/src/main/scala/scalafix/test/cli/Rules.scala

```diff
@@ -1,6 +1,6 @@
 package scalafix.test.cli
 
-import scala.meta._
```

Comment:
https://github.com/scalacenter/scalafix/actions/runs/7767691536/job/21184796798?pr=1800
```
[error] --- /home/runner/work/scalafix/scalafix/scalafix-tests/integration/src/main/scala/scalafix/test/cli/Rules.scala
[error] +++ <expected fix>
[error] @@ -1,7 +1,5 @@
[error]  package scalafix.test.cli
[error]  
[error] -import scala.meta._
[error] -
[error]  import metaconfig.Configured
[error]  import scalafix.internal.util.PositionSyntax._
[error]  import scalafix.internal.util.SymbolOps
...
[error] (integration2_13 / scalafixAll) scalafix.sbt.ScalafixFailed: TestError
```

Reply:
This is clearly a side effect of this PR as [we are dogfooding](https://github.com/scalacenter/scalafix/pull/1784). The weird thing is that I can't reproduce locally.

Reply:
Relaxing the `isUnused()` implementation exposed a pretty serious bug: the mutable collections used in `OrganizeImports` are not cleared from one file to another, as the class is instantiated only once per scalafix invocation.

https://github.com/scalacenter/scalafix/blob/9271030ca37e4af39f416ce9fe1f7fe1560c080e/scalafix-rules/src/main/scala/scalafix/internal/rule/OrganizeImports.scala#L65-L69

I'll fix it in a separate PR, and rebase this PR to remove the need for the hacky second commit.